### DISCUSSION TITLE
Fix N+1 problem in GraphQL federation

### DISCUSSION THREAD

**Discussion by Sheikhharis50** [strawberry-graphql/strawberry#1952] [General] [answered]
Hi guys,

I have pasted a link below where I have explained the issue, I would like to discuss related to the N+1 problem in the Federation.
I can also provide an example of it.

**Issue comment**
https://github.com/strawberry-graphql/strawberry/issues/115#issuecomment-1152992970

**Example Repository for federation (take a look at ProductType)**
https://github.com/Sheikhharis50/graphql-federation/blob/master/product-service/app_product/types.py

Moreover, @patrick91 has mentioned related to async `resolver_reference` in the below issue, we might also need that.
https://github.com/strawberry-graphql/strawberry/issues/591

**Comment by patrick91** [comment]
I think adding support for async resolver_reference shouldn't be too difficult, @Sheikhharis50 do you want to try?

**Reply by Sheikhharis50** [reply]
Yes, I would love to, can you guide me through it?

**Reply by patrick91** [reply]
@Sheikhharis50 sorry for disappearing, do you still want to work on this?

**Reply by Sheikhharis50** [reply]
Yes, can you guide me how can i contribute?

**Reply by patrick91** [reply]
sure, are you on discord? we can chat there 😊 I spent a bit of time investigating this and it's not that simple unfortunately 😊

**Reply by Sheikhharis50** [reply]
yes, I use discord, that's my id "sheikhhariszahid#3299".

**Comment by notypecheck** [comment, marked as answer]
I'm probably a bit too late but you can use dataloaders to solve N+1: https://gitlab.com/manga-microservices/identity/-/blob/main/src/gql/users/types.py#L18

**Reply by patrick91** [reply]
wait, we support `async resolve_reference`?

**Reply by notypecheck** [reply]
Hey, it works 🤷

**Reply by patrick91** [reply]
oh, I didn't realise that 🤦‍♂️ maybe we need to add a test for this then

**Reply by patrick91** [reply]
finally took some time to check this, and yes, it is supported thanks to graphql-core resolve value. I'll make a PR to add a test so we don't break this support in future :D
